applications/crossbar/src/modules/cb_fmc_devices.erl

```diff
@@ -0,0 +1,285 @@
+    ResultDoc1 = wh_doc:public_fields(ResultDoc),
+    Context1 = cb_context:set_doc(Context, ResultDoc1),
+    cb_context:set_resp_data(Context1, ResultDoc1).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% If the HTTP verb is POST, execute the actual action, usually a db save
+%% (after a merge perhaps).
+%% @end
+%%--------------------------------------------------------------------
+-spec post(cb_context:context(), path_token()) -> cb_context:context().
+post(Context, Id) ->
+    AccountId = cb_context:account_id(Context),
+    Doc1 = wh_json:set_values([{<<"_id">>, Id}
+        ,{<<"pvt_type">>, <<"fmc_device">>}
+        ,{<<"account_id">>, AccountId}]
+        ,cb_context:doc(Context)),
+    DeviceId = wh_json:get_value(<<"device_id">>, Doc1),
+    {'ok', DeviceDoc} = couch_mgr:open_cache_doc(cb_context:account_db(Context), DeviceId),
+    ANumber = wh_json:get_value([<<"call_forward">>, <<"number">>], DeviceDoc),
+    Doc2 = wh_json:set_value(<<"a_number">>, ANumber, Doc1),
+    {'ok', ResultDoc} = couch_mgr:save_doc(?WH_FMC_DB, Doc2),
+    ResultDoc1 = wh_doc:public_fields(ResultDoc),
+    Context1 = cb_context:set_doc(Context, ResultDoc1),
+    cb_context:set_resp_data(Context1, ResultDoc1).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% If the HTTP verb is DELETE, execute the actual action, usually a db delete
+%% @end
+%%--------------------------------------------------------------------
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+delete(Context, Id) ->
+    {'ok', _} = couch_mgr:del_doc(?WH_FMC_DB, Id),
+    cb_context:set_resp_status(Context, 'success').
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec on_successful_validation('undefined' | api_binary(), cb_context:context()) -> cb_context:context().
+on_successful_validation('undefined', Context) ->
+    FMCValue = wh_json:get_value(<<"x_fmc_value">>, cb_context:doc(Context)),
+    DeviceId = wh_json:get_value(<<"device_id">>, cb_context:doc(Context)),
+    case couch_mgr:open_cache_doc(cb_context:account_db(Context), DeviceId) of
+        {'ok', DeviceDoc} ->
+            Doc = wh_json:set_value(<<"pvt_type">>, <<"fmc_device">>, cb_context:doc(Context)),
+            DeviceType = wh_json:get_value(<<"device_type">>, DeviceDoc),
+            Number = wh_json:get_value([<<"call_forward">>, <<"number">>], DeviceDoc),
+            case DeviceType of
+                <<"cellphone">> ->
+                    case is_existing_fmc(Number, FMCValue, 'undefined') of
+                        'true' ->
+                            cb_context:add_validation_error(<<"fmc">>, <<"unique">>, wh_json:from_list([{<<"message">>, <<"FMC Device with this A-Number and FMC Value already exists">>}]), Context);
```

Comment:
Same for all these error responses, make a function that calls add_validation_error please